PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,468 @@
+                that are created when you use the -match operator. This isn't as flexible nor as
+                robust as using [regex]::Matches(), but it's handy in a pinch.
+                
+                In this case, you'll see that the "sh" in "shears" weren't part of the match. The
+                pattern "zero or more p's, followed by e a r s" is found in the string, even though
+                there are parts of the string that aren't relevant to that matching effort.
+    
+                $matches[0] is always your most recent match.
+            #>
+    
+            '____' | Should -Be $matches[0]
+        }
+    }
+
+    Context '+' {
+        It 'specifies 1 or more of something' {
+            <#
+                The + symbol is a lot like the star, but instead of matching zero or more, it
+                matches one or more.
+            #>
+    
+            '____' -match 'p+ickles' | Should -BeTrue
+            '____' -match 'p+ickles' | Should -BeFalse
+        }
+    }
+
+    Context '?' {
+        It 'specifies 0 or 1 of something' {
+            # The ? matches "zero or one" of something.
+            
+            [bool]$boatMatch = $____ # Should be either $true or $false
+            $boatMatch | Should -Be 'flying through the sky' -match 'f+lying'
+            
+            [bool]$floatMatch = $____ # Should be either $true or $false
+            $floatMatch | Should -Be 'floating away' -match 'b+oat'
+        }
+    }
+
+    Context 'Greediness and Laziness' {
+        It 'matches a little or matches a lot' {
+            <#
+            A number of the quantifiers have two versions: A greedy version which tries to match an
+            element as many times as possible, and a non-greedy (or lazy) version which tries to
+            match an element as few times as possible. You can turn a greedy quantifier into a lazy
+            quantifier by simply adding a ?.
+            #>
+
+            $greedy = 'o+'
+            $lazy = 'o+?'
+            '____' | Should -Be @([regex]::Matches('helloooooo', $greedy).Value)[0]
+            '____' | Should -Be @([regex]::Matches('helloooooo', $lazy).Value)[0]
+        }
+    }
+}
+
+Describe 'Special Symbols' {
+    <#
+        In regex, there are far more symbols with unique meanings than just quantifiers. What comes
+        next is not a definitive guide to EVERY SINGLE ONE, but rather an introduction to some of
+        the most common and useful special symbols in regex.
+        
+        This is ^\where things st\Art getting \weir\D$
+    #>
+    
+
+    Context '. (period)' {
+        It 'matches any character' {
+            # The . (period) matches literally any character
+    
+            [bool]$morningMatch = $____ # Should be either $true or $false
+            $morningMatch | Should -Be 'Lazy Sunday mornings' -match 'S.nday'
+    
+            '____' -match 'invi.a.ion' | Should -BeTrue
```

Comment:
These can be separate `It` blocks, and that first one needs to be refactored similarly to above. 🙂